Added test for regression bug, closes #1564

winterismute opened the pull request:
Addresses https://github.com/HaxeFlixel/flixel/issues/1564

Gama11 commented on tests/unit/src/flixel/FlxObjectTest.hx:
```diff
@@ -74,58 +74,72 @@ class FlxObjectTest extends FlxTest
 	{
-		var object1 = new FlxObject(0, 0, 10, 10);
-		var object2 = new FlxObject(20, 2, 10, 6);
+		var object1:FlxObject = new FlxObject(0, 0, 10, 10);
```
Not a fan of adding the `:FlxObject` explicitly here. That's where Haxe's type inference comes in handy and helps to avoid redundancy (see in the "Leveraging Type Inference" section [here](http://haxeflixel.com/documentation/code-style/)).

Gama11 commented:
If I go back to 1eaf0fedc6d3c4e9e58249d259ca8d6f1b9dc391 (just before my fix), this test is not failing as it should.

winterismute commented:
Strange, I did check that before pushing... I will look into this soon.

winterismute commented:
I have updated the branch, you were right, I made a mistake when double-checking (put AND instead of OR). I have changed the test with a different one which involves a tilemap, which seems to be the main (if not only) case in which it could go wrong (which was btw also the reported case).
I have also changed those styling things. Hope now it's fine.

Gama11 commented:
Thanks!
